### PR TITLE
resolve #38: close gpg connection on exit

### LINKED ISSUE #38: new wsl2-ssh-pageant.exe instance is create for every gpg communication
**Describe the bug**
This is how I start the process in WSL 2:

```
setsid socat UNIX-LISTEN:$GPG_AGENT_SOCK,fork EXEC:"$HOME/.ssh/wsl2-ssh-pageant.exe --verbose --gpgConfigBasepath 'C:/Users/user-name/AppData/Local/gnupg' --gpg S.gpg-agent" >/dev/null 2>&1 &
```

**To Reproduce**
1. Start a WSL 2 terminal
2. Start the `socat` process like above
3. Repeat `gpg --card-status` 3 times
4. Check the `wsl2-ssh-pageant.exe` process in Windows Task Manager

**Expected behavior**

No `wsl2-ssh-pageant.exe` process should be found in the Task Manager

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: Windows 11

**Additional context**

- Gpg4win v4.0.0
- wsl2-ssh-pageant v1.4.0

### PR DESCRIPTION
**Pull request by davidshen84**

### CODE REVIEW

**Comment by davidshen84** [comment]
- [ ] remove `-H=windowsgui` build option

File: main.go [hunk 1]
```diff
@@ -141,15 +141,16 @@ func main() {
...
+		configDir, err := os.UserConfigDir()
 		if err != nil {
-			log.Fatal("failed to find user home dir")
+			log.Fatal("failed to find user config dir")
 		}
 		basePath := *gpgConfigBasepath
 		// fallback to default location if not specified
 		if basePath == "" {
-			basePath = filepath.Join(homeDir, "AppData", "Roaming", "gnupg")
+			basePath = filepath.Join(configDir, "gnupg")
```
Comment (kaeltis) [inline main.go:151]: According to https://pkg.go.dev/os#UserConfigDir `os.UserConfigDir()` just resolves to %appdata% on Windows,

so shouldn't this be `basePath = filepath.Join(configDir, "Local", "gnupg")`?
(or `Roaming`, depending on the gpg version, see https://github.com/BlackReloaded/wsl2-ssh-pageant/issues/33#issuecomment-1005743099)

Comment (kaeltis) [inline main.go:151]: According to https://pkg.go.dev/os#UserConfigDir `os.UserConfigDir()` just resolves to %appdata% on Windows,

so shouldn't this be `basePath = filepath.Join(configDir, "Local", "gnupg")`?
(or `Roaming`, depending on the gpg version, see https://github.com/BlackReloaded/wsl2-ssh-pageant/issues/33#issuecomment-1005743099)

Comment (davidshen84) [reply main.go:151]: - `%AppData%` resolves to `C:\Users\user\AppData\Roaming`
- `%LocalAppData%` resolves to `C:\Users\user\AppData\Local`

So by default, we use the `gnupg` folder under the `%AppData%` directory.

Comment (kaeltis) [reply main.go:151]: True, completely forgot about `%LocalAppData%` being a thing, my bad!

**Comment by davidshen84** [comment]
It seems the *ghost* process is caused solely by the `-H=windowsgui` build option and has nothing to do with `gpgConn` instance. Nevertheless, I think it is still a good idea to explicitly close a resource.

**Comment by kaeltis** [comment]
`-H=windowsgui` was apparently added in https://github.com/BlackReloaded/wsl2-ssh-pageant/commit/937e3d8e82ae71a73f6ec6a390e00e91776efa1a due to the bug mentioned in https://github.com/BlackReloaded/wsl2-ssh-pageant/issues/30 - however I can't reproduce it using the `Windows Subsystem for Linux Preview` from the store, even without the `-H=windowsgui` - so maybe it was fixed by Microsoft and can be removed again?

If so, it might be best to revert the change in the workflow file https://github.com/BlackReloaded/wsl2-ssh-pageant/pull/39/files#diff-678682767f2477de3e3c546746f8568b9a1942b2c647d32331d7e774b8ff8d9fL20 and just update https://github.com/BlackReloaded/wsl2-ssh-pageant/blob/main/Makefile#L3=

**Comment by davidshen84** [comment]
I updated the workflow already and I think it is still better to close the
gpg connection explicitly.

**Pull request closed without merging**